dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java

```diff
@@ -290,6 +335,31 @@ public void run()
         }
     }
 
+    /**
+     * Returns RestartInformation for the given streamPos if possible.
+     * 
+     * @return RestartInformation
+     */
+    RestartInformation getRestartInformationForStreamPos(long streamPos)
+    {
+        RestartInformation restartInfo = new RestartInformation();
+
+        // Note if we don't find a valid start point we will restart from the beginning of the file
+        for (int i = 0; i < streamPosOfFetches.size(); i++)
```

Comment:
It seems we can simplify this for loop condition having a lone index, and having it decrement.
At the very least, let's move the streamPosOfFetches.size() calls out of the loop condition and body onto a local var.